pyuvsim/simsetup.py

```diff
@@ -0,0 +1,388 @@
+
+    if not nt:
+        if not it:
+            raise ValueError("Either integration_time or Ntimes " +
+                             "must be included in parameters:" + kws_used)
+        if st and et:
+            time_params['duration'] = time_params['start_time'] - time_params['end_time']
+            dd = True
+        if dd:
+            time_params['Ntimes'] = int(np.floor(time_params['duration'] /
+                                        (time_params['integration_time']
+                                        * dayspersec))) + 1
+        else:
+            raise ValueError("Either duration or time bounds " +
+                             "must be specified: " + kws_used)
+
+    if not it:
+        if not dd:
+            raise ValueError("Either duration or integration time " +
+                             "must be specified: " + kws_used)
+        time_params['integration_time'] = (time_params['duration'] /
+                                           float(time_params['Ntimes']))
+
+    if not dd:
+        time_params['duration'] = time_params['integration_time'] * time_params['Ntimes']
+    if not st:
+        if et and dd:
+            time_params['start_time'] = time_params['end_time'] - time_params['duration']
+    if not et:
+        if st and dd:
+            time_params['end_time'] = time_params['start_time'] + time_params['duration']
+    if not (st or et):
+        raise ValueError("Either a start or end time must be specified" + kws_used)
+
+    time_arr = np.linspace(time_params['start_time'],
+                           time_params['end_time'] +
+                           time_params['integration_time']*dayspersec,
+                           time_params['Ntimes'])
+
+    Nbl = (param_dict['Nants_data']+1)*param_dict['Nants_data'] / 2
+
+    time_arr = np.sort(np.tile(time_arr, Nbl))
+    param_dict['integration_time'] = time_params['integration_time']
+    param_dict['time_array'] = time_arr
+    param_dict['Ntimes'] = time_params['Ntimes']
+    del param_dict['time']
+
+    # Now make a UVData object with these settings built in.
+    # The syntax below allows for other valid uvdata keywords to be passed
+    #  without explicitly setting them here.
+
+    uv_obj = UVData()
+    print param_dict
+    for k in param_dict:
+        print k, param_dict[k]
+        if hasattr(uv_obj, k):
+            setattr(uv_obj, k, param_dict[k])
+
+    bls = np.array([uv_obj.antnums_to_baseline(j, i)
+                   for i in range(0, uv_obj.Nants_data)
+                   for j in range(i, uv_obj.Nants_data)])
+
+    uv_obj.baseline_array = np.tile(bls, uv_obj.Ntimes)
+    uv_obj.Nbls = bls.size
+    uv_obj.Nblts = uv_obj.Nbls * uv_obj.Ntimes
+
+    uv_obj.ant_1_array, uv_obj.ant_2_array = \
+        uv_obj.baseline_to_antnums(uv_obj.baseline_array)
+
+    return uv_obj, beam_list, beam_ids
+
+
+def uvfits_to_telescope_config(file_in, beam_filepath, layout_csv_name=None,
```

Comment:
Why is this specific to uvfits? Should we change it to take a UVData object so that it can be based on any file type?

Reply:
Good idea. I'll switch that out.